lib/internal/histogram.js

```diff
@@ -1,38 +1,72 @@
 'use strict';
 
+const {
+  NumberIsNaN,
+  NumberIsInteger,
+  NumberMAX_SAFE_INTEGER,
+  ObjectSetPrototypeOf,
+  SafeMap,
+  Symbol,
+  TypeError,
+} = primordials;
+
+const {
+  Histogram: _Histogram
+} = internalBinding('performance');
+
 const {
   customInspectSymbol: kInspect,
 } = require('internal/util');
 
-const { format } = require('util');
-const { NumberIsNaN, SafeMap, Symbol } = primordials;
+const { inspect } = require('util');
 
-const { ERR_INVALID_ARG_VALUE } = require('internal/errors').codes;
-const { validateNumber } = require('internal/validators');
+const {
+  codes: {
+    ERR_INVALID_ARG_VALUE,
+    ERR_INVALID_ARG_TYPE,
+    ERR_OUT_OF_RANGE,
+  },
+} = require('internal/errors');
+
+const {
+  validateNumber,
+} = require('internal/validators');
 
 const kDestroy = Symbol('kDestroy');
 const kHandle = Symbol('kHandle');
+const kMap = Symbol('kMap');
+
+const {
+  kClone,
+  kDeserialize,
+  JSTransferable,
+} = require('internal/worker/js_transferable');
 
-// Histograms are created internally by Node.js and used to
-// record various metrics. This Histogram class provides a
-// generally read-only view of the internal histogram.
-class Histogram {
-  #map = new SafeMap();
+class Histogram extends JSTransferable {
+  [kMap] = new SafeMap();
 
   constructor(internal) {
+    super();
     this[kHandle] = internal;
   }
```

Comment:
nit: `[kMap]` assignment could be move inside the constructor so all properties are defined in one place.